department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-13368

Resolves [APPEALS-13368](https://vajira.max.gov/browse/APPEALS-13368)

### Description
Created the logic to cancel a post initial letter task and show the hold period of the post initial letter task within the case timeline. Also, refactored parts of the code to remove the TimedHoldTask methods from the post Initial task letter task class.

### Acceptance Criteria
- [ ] Cancel action cancels the PostSendInitialNotificationLetterHoldingTask
- [ ] When cancelled, the PostSendInitialNotificationLetterHoldingTask displays the correct days on hold per the AC

### Testing Plan
1. To create a Contested Claim appeal, go to the login page and sign in as a BVA Intake user like BVADWISE. If you have already have Contested Claim appeals, you can skip to step 7.
2. Navigate to Intake
3. Select the first option from the dropdown and **continue** button.
![image](https://user-images.githubusercontent.com/99915461/220455078-29e62854-0b9d-424a-abb6-5464b389ca7b.png)

4. Enter a veteran file number and click the magnifying glass to continue ( I used 400000004).
![image](https://user-images.githubusercontent.com/99915461/220455180-8a054f6e-3b78-411f-b24d-f8f69c372684.png)

5. Fill out the form and click the continue button.
![image](https://user-images.githubusercontent.com/99915461/220455319-c5c396e1-9fa8-47aa-b9f7-2171793d03c9.png)

6. Click **add issues** and select a Contested Claim issue. Click **establish appeal** to continue. You should be greeted by a green success banner.
![image](https://user-images.githubusercontent.com/99915461/220455424-24f75757-1619-40d6-b26f-cd70512e9437.png)
![image](https://user-images.githubusercontent.com/99915461/220455508-b73b9fd0-fe9e-4c74-91ff-670a6950f634.png)

7. Return to the Sign In page and sign in as a COB user. 
![image](https://user-images.githubusercontent.com/99915461/220455778-431e65ce-31ef-4a65-a5ec-2cd9fff77467.png)

8. Navigate to **Queue**. 
9. Click the **switch views** dropdown and select **Clerk of Board team cases**.
![image](https://user-images.githubusercontent.com/99915461/220455949-f5f45a57-66ad-410c-9bf3-526c3ba76656.png)
![image](https://user-images.githubusercontent.com/99915461/220455996-a814e7ab-e853-4434-8ab6-54242c2d27b5.png)

10. You should see the Send Initial Notification task that was created from the intake (or existing seed data).
![image](https://user-images.githubusercontent.com/99915461/220456132-c2673a41-959f-40c3-b8c2-7c0413055fa3.png)

11. Click on a link to navigate to the case details page.
12. On the Case Details page, select the **Actions** dropdown and click **Mark task complete**.
![image](https://user-images.githubusercontent.com/99915461/220456340-874c6d30-cbbc-42db-b41f-ffc6696ca83a.png)

13. On the next menu, select **45 days** and **mark as complete**.
![image](https://user-images.githubusercontent.com/99915461/220456455-3792861a-1e9e-4b1b-a770-f66bcda7afdf.png)

14. You should navigate to the Organization Queue Table and be greeted with a success banner.
![image](https://user-images.githubusercontent.com/99915461/220456570-49e496c5-6737-46ab-93df-058acf56dd1f.png)

15. Click on the **On Hold** tab on the page. You should see the newly created Post Initial Task and a column showing how many days it is on hold (bottom 2 tasks in the image)
![image](https://user-images.githubusercontent.com/99915461/220456716-5289957b-97bb-4f2c-a602-b275ea9bf9ec.png)

16. Click on the task to navigate to the Case Details page for that appeal. Copy the veteran file number to navigate back to the page on step 20.
17. On the Case Details page, click the **action** dropdown and select **Cancel task**.
![image](https://user-images.githubusercontent.com/99915461/220456983-5171e112-b085-4af2-ac86-542ea1b0c4c9.png)

18. Fill out the instructions on the pop-up modal and click the **submit** button.
![image](https://user-images.githubusercontent.com/99915461/220457088-8b1f4657-af9c-4479-9739-989f46ef0f83.png)

19. You should be greeted with a green success banner. 
![image](https://user-images.githubusercontent.com/99915461/220457149-af0b7567-17a1-482e-a8fc-ef1bc8ca1e24.png)

20. Navigate back to the Case Details page using the back arrow or by clicking the **case search** icon and entering the veteran file number from step 16.
21. On the Case Timeline at the bottom of the page, you should see the Post Initial task has been cancelled. Click the **view task instructions** link to make sure the On Hold time is showing properly: 
![image](https://user-images.githubusercontent.com/99915461/220457882-bd2b7402-da68-4630-b94a-9085a9b8d877.png)
![image](https://user-images.githubusercontent.com/99915461/220458074-20f09633-bf0f-4989-a12a-fb7ee1b29efb.png)
